multitenant: move tenantcostmodel

RaduBerinde opened the pull request:
Move `tenantcostmodel` from `pkg/util` to the new `pkg/multitenant`.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/68260)
<!-- Reviewable:end -->

RaduBerinde committed:
multitenant: move tenantcostmodel

Move `tenantcostmodel` from `pkg/util` to the new `pkg/multitenant`.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/68260#-:-Mg6UsNB9SKVMBhi-bwq:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
